src/librustc_errors/emitter.rs

```diff
@@ -447,7 +447,7 @@ pub trait Emitter {
             .copied()
             .chain(span.span_labels().iter().map(|sp_label| sp_label.span))
             .filter_map(|sp| {
-                if !sp.is_dummy() && sm.span_to_filename(sp).is_macros() {
+                if !sp.is_dummy() && sm.is_imported(sp) {
```

Comment:
At the very least, `<*macros>` in comments should be replaced with "external macros" or so.

Reply:
Done.